### PR TITLE
NIFI-2610 added initial support for categorizing tests

### PR DESCRIPTION
**Pull request by olegz**

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/controller/scheduling/TestProcessorLifecycle.java [hunk 1]
```diff
@@ -191,6 +193,7 @@ public void validateStopCallsAreMeaninglessIfProcessorNotStarted() throws Except
      * operations can only be @OnScheduled, @OnUnscheduled, @OnStopped.
      */
     @Test
+    @Category(IntegrationTest.class)
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/controller/scheduling/TestProcessorLifecycle.java]: This test (and the one following) were not marked as @Ignore before and aren't in a named integration test file (starting or ending in IT). Are these actually integration tests that have been running as unit tests, or do they just take too long to run, or are they issuing failures? In the last case, I recommend against labelling failing tests with the Category and keep the @Ignore so we can easily find them in source without having to run all (integration) tests to see which pass and which fail.

File: nifi-api/src/main/java/org/apache/nifi/testutil/IntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.testutil;
+
+import org.junit.experimental.categories.Category;
```
Comment (mattyb149) [inline nifi-api/src/main/java/org/apache/nifi/testutil/IntegrationTest.java]: I haven't compiled yet but I can't see where (except for Javadoc) the Category class is used. Does this cause checkstyle errors, or does Javadoc need it as an import or something?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/controller/scheduling/TestProcessorLifecycle.java (hunk 1 above)
Comment (olegz) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/controller/scheduling/TestProcessorLifecycle.java]: These are the tests that deal with concurrency which is hard to tune when are running under load of full build. So these annotation will exclude them from full build yet they are still valid JUnit tests and will run under JUnit as before essentially sparing you from commenting/uncommenting @Ignore.

Comment (olegz) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/controller/scheduling/TestProcessorLifecycle.java]: Forgot to mention. The two tests that you see here (as an example) were failing sporadically, due to unexpected delays related to the load of the full build.

File: nifi-api/src/main/java/org/apache/nifi/testutil/IntegrationTest.java (hunk 2 above)
Comment (olegz) [inline nifi-api/src/main/java/org/apache/nifi/testutil/IntegrationTest.java]: Just javadoc

**Comment by olegz** [comment]
Due to Travis failure, I am pulling this PR and will @Ignore the failing tests instead. I'll also raise a separate JIRA to address Travis issue in the future.

**Pull request closed without merging**